### ISSUE TITLE
Not finding vhosts on A{mazon,rch}Linux

### ISSUE THREAD

**Issue by bmw** [certbot/certbot#4905]
## My operating system is (include version):

Amazon Linux 2017.03

## I installed Certbot with (certbot-auto, OS package manager, pip, etc):

certbot-auto

## I ran this command and it produced this output:

```
sudo env PATH=$PATH ./certbot-auto --apache --debug -v -d example.org
...
Could not reverse map the HTTPS VirtualHost to the original
...
```

## Certbot's behavior differed from what I expected because:

Certbot should be able to find the vhost.

This issue was created in response to https://community.letsencrypt.org/t/rate-limited-after-trying-multiple-solutions-to-could-not-reverse-map-the-https-virtualhost-to-the-original-issue/37252. Adding `--apache-vhost-root` causes the "Could not reverse map..." error message to go away and be replaced by:

> No vhost exists with servername or alias of: example.org (or it's in a file with multiple vhosts, which Certbot can't parse yet). No vhost was selected. Please specify ServerName or ServerAlias in the Apache config, or split vhosts into separate files.

I played with this myself for a bit in Docker using http24 installed in the default paths and even with `--apache-vhost-root`, we're not finding the vhost. The return value from `certbot_apache.configurator.get_virtual_hosts()` is an empty list.

@joohoi, since Amazon Linux is Red Hat based, this might be a good test to see how better support for non-Debian systems is coming in Apache.

**Comment by joohoi** [comment]
This should be fixed in #4104 as the issue seems to be in make_vhost_ssl

**Comment by bmw** [comment]
I believe another instance of this problem can be found in https://community.letsencrypt.org/t/archlinux-how-to-configur-lets-encrypt/39258/32. At the very least, #4104 seems to fix the issue.
